src/components/UserList.tsx

```diff
@@ -60,14 +60,8 @@ class UserList extends React.PureComponent<UserListProps> {
             [classes.canManageUsers]: canManageUsers,
           })}
         >
-          {canManageUsers && <ManageUsersView removeUser={removeUser} users={users} />}
+          <ManageUsersView removeUser={canManageUsers ? removeUser : undefined} users={users} />
           {users.length === 0 && <Typography className={classes.noUsers}>(None)</Typography>}
-          {!canManageUsers &&
-            users.map((user) => (
-              <Typography noWrap key={user}>
-                {user}
-              </Typography>
-            ))}
```

Comment:
Note for reviewers: If the user cannot manage users, then instead of just displaying the emails/groups in plain text, we will still display the users in a "chip" format. It just won't have the little "x" to delete the group. 

**Before** this change:
<img width="316" alt="image" src="https://github.com/DataBiosphere/jade-data-repo-ui/assets/13254229/59c0f69c-d20c-4c50-b3da-67b62ab01214">

**After** this change:
<img width="427" alt="image" src="https://github.com/DataBiosphere/jade-data-repo-ui/assets/13254229/2171f752-9e2c-4fb8-892f-fe6230cbc67f">

Reply:
Nice!

Will you remind me, what's the case when a user will see this tab, but not be able to manage users?  Is that when a user is a snapshot custodian but not a steward?

Reply:
As far as I can tell, there isn't a scenario where a user can see the membership on dataset/snapshot but not edit it. 

However, I'm using this setting to disable all users' ability to edit the auth domain groups. 

____
More on the first point:
- We only show the Roles & membership tab if the user is a [steward on the snapshot](https://github.com/DataBiosphere/jade-data-repo-ui/blob/develop/src/components/snapshot/overview/SnapshotOverviewPanel.tsx#L121), so they will be able to manage users
- We also reference this code when sharing a snapshot, but in that case it's in "create mode" no matter the user role, so they will always be able to manage users.